src/webapp/components/survey/hook/useSurveyForm.ts

```diff
@@ -21,6 +22,17 @@ export function useSurveyForm(formType: SURVEY_FORM_TYPES, eventId: string | und
     } = useCurrentSurveys();
     const [error, setError] = useState<string>();
 
+    const shouldDisableSave = (): boolean => {
+        if (!questionnaire) return true;
+        const allQuestions = questionnaire.stages.flatMap(stage => {
+            return stage.sections.flatMap(section => {
+                return section.questions.map(question => question);
+            });
+        });
+
+        return allQuestions.some(question => question.errors.length > 0);
```

Comment:
Not a terribly complex logic, but the abstraction can be typically created in the entity, with a nice name.